src/pytorch_lightning/trainer/trainer.py

```diff
@@ -628,12 +628,11 @@ def _setup_on_init(self) -> None:
         self.num_sanity_val_batches = []
         self.num_test_batches = []
         self.num_val_batches = []
+        self.num_predict_batches = []
         self.test_dataloaders = None
         self.val_dataloaders = None
-        self._last_train_dl_reload_epoch = float("-inf")
-        self._last_val_dl_reload_epoch = float("-inf")
-
-        self.num_predict_batches = []
+        self.predict_dataloaders = None
+        self._last_val_dl_reload_epoch: Optional[int] = None
```

Comment:
we set this to None to make sure if validation is triggered, we reload the dataloaders regardless.